src/scripts/main.js

```diff
@@ -1,3 +1,338 @@
+        continue;
+      }
+      moveCell(gameField, i, j, -1, 0);
+    }
+  }
+}
+
+function mergeUp(gameField) {
+  for (let i = 0; i < rowLength - 1; i++) {
+    for (let j = 0; j < columnLength; j++) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+
+      if (gameField[i][j] === gameField[i + 1][j]) {
+        gameField[i][j] = gameField[i][j] * 2;
+        score += gameField[i][j];
+        gameField[i + 1][j] = 0;
+      }
+    }
+  }
+}
+
+function moveDown(gameField) {
+  for (let i = rowLength - 1; i >= 0; i--) {
+    for (let j = 0; j < columnLength; j++) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+      moveCell(gameField, i, j, 1, 0);
+    }
+  }
+}
+
+function mergeDown(gameField) {
+  for (let i = rowLength - 1; i > 0; i--) {
+    for (let j = 0; j < columnLength; j++) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+
+      if (gameField[i][j] === gameField[i - 1][j]) {
+        gameField[i][j] = gameField[i][j] * 2;
+        score += gameField[i][j];
+        gameField[i - 1][j] = 0;
+      }
+    }
+  }
+}
+
+function moveRight(gameField) {
+  for (let i = 0; i < rowLength; i++) {
+    for (let j = columnLength - 1; j >= 0; j--) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+      moveCell(gameField, i, j, 0, 1);
+    }
+  }
+
+  return gameField;
+}
+
+function mergeRight(gameField) {
+  for (let i = 0; i < rowLength; i++) {
+    for (let j = columnLength - 1; j > 0; j--) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+
+      if (gameField[i][j] === gameField[i][j - 1]) {
+        gameField[i][j] = gameField[i][j] * 2;
+        score += gameField[i][j];
+        gameField[i][j - 1] = 0;
+      }
+    }
+  }
+}
+
+function moveLeft(gameField) {
+  for (let i = 0; i < rowLength; i++) {
+    for (let j = 0; j < columnLength; j++) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+      moveCell(gameField, i, j, 0, -1);
+    }
+  }
+}
+
+function mergeLeft(gameField) {
+  for (let i = 0; i < rowLength; i++) {
+    for (let j = 0; j < columnLength; j++) {
+      if (!gameField[i][j]) {
+        continue;
+      }
+
+      if (gameField[i][j] === gameField[i][j - 1]) {
+        gameField[i][j] = gameField[i][j] * 2;
+        score += gameField[i][j];
+        gameField[i][j - 1] = 0;
+      }
+    }
+  }
+}
+
+function move(e) {
+  if (!button.classList.contains('restart')) {
+    return;
+  }
+
+  const fieldCopy = field.map(function(arr) {
+    return arr.slice();
+  });
+
+  switch (e.key) {
+    case 'ArrowUp':
+      moveUp(field);
+      mergeUp(field);
+      moveUp(field);
+      break;
+
+    case 'ArrowDown':
+      moveDown(field);
+      mergeDown(field);
+      moveDown(field);
+      break;
+
+    case 'ArrowRight':
+      moveRight(field);
+      mergeRight(field);
+      moveRight(field);
+      break;
+
+    case 'ArrowLeft':
+      moveLeft(field);
+      mergeLeft(field);
+      moveLeft(field);
+      break;
+  }
```

Comment:
Don't forget to add a default case to handle situations when key doesn't match any of cases